pkg/assembler/backends/testing/osv.go

```diff
@@ -17,77 +17,181 @@ package testing
 
 import (
 	"context"
+	"fmt"
+	"log"
+	"strconv"
 	"strings"
 
 	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+	"github.com/vektah/gqlparser/v2/gqlerror"
 )
 
+// TODO: convert to unit test
 func registerAllOSV(client *demoClient) {
-	client.registerOSV("CVE-2019-13110")
-	client.registerOSV("CVE-2014-8139")
-	client.registerOSV("CVE-2014-8140")
-	client.registerOSV("CVE-2022-26499")
-	client.registerOSV("GHSA-h45f-rjvw-2rv2")
-}
-
-// Ingest OSV
+	ctx := context.Background()
 
-func (c *demoClient) registerOSV(id string) *model.Osv {
-	idLower := strings.ToLower(id)
-	for i, o := range c.osv {
-		c.osv[i] = registerOsvID(o, idLower)
-		return c.osv[i]
+	inputs := []model.OSVInputSpec{{
+		OsvID: "CVE-2019-13110",
+	}, {
+		OsvID: "CVE-2014-8139",
+	}, {
+		OsvID: "CVE-2014-8140",
+	}, {
+		OsvID: "CVE-2022-26499",
+	}, {
+		OsvID: "GHSA-h45f-rjvw-2rv2",
+	}}
+	for _, input := range inputs {
+		_, err := client.IngestOsv(ctx, &input)
+		if err != nil {
+			log.Printf("Error in ingesting: %v\n", err)
+		}
 	}
+}
 
-	newOsv := &model.Osv{}
-	newOsv = registerOsvID(newOsv, idLower)
-	c.osv = append(c.osv, newOsv)
+const osv string = "osv"
 
-	return newOsv
+// Internal data: osv
+type osvMap map[string]*osvNode
+type osvNode struct {
+	id     uint32
+	osvIDs osvIDMap
+}
```

Comment:
This is missing the string key.

Reply:
Which string key?

Reply:
The key that is used in the map above. If you have `map[string]*Smth` then in `type Smth struct` we need a string field to point back to the key, so we can go back up (given a pointer, find the string that is used as a key).

If we don't need to index on the string, then we don't need the map either, it's just a list.

Compare https://github.com/mihaimaruseac/guac/blob/9d9c9be4c18a1f2645353867386d81ca26e20970/server/backends/memory/trees.go#LL33-L38C2 (where we have map and string key) with https://github.com/mihaimaruseac/guac/blob/9d9c9be4c18a1f2645353867386d81ca26e20970/server/backends/memory/trees.go#L102-L108 where we don't have map, just a list

Reply:
Makes more sense to make it a list as it will be a single key. I will change that.

Reply:
On second thought it might be better to add the type as the key. In case we decide to add more types in the future. I will update.